hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/joiner/DefaultLabelJoinerBuilder.java

```diff
@@ -91,5 +214,106 @@ private Label getLabelFromMap(String labelId, Map<String, Label> labelMap) {
       }
       return label;
     }
+
+    private Single<EntityRequest> buildEntityRequestForType(
+        GraphQlRequestContext context,
+        String entityType,
+        String labelId,
+        int limit,
+        DataFetchingFieldSelectionSet selectionSet) {
+      return buildLabelIdFilter(context, entityType, labelId)
+          .flatMap(
+              filterArguments ->
+                  buildEntityRequest(context, entityType, limit, filterArguments, selectionSet));
+    }
+
+    private Single<EntityRequest> buildEntityRequest(
+        GraphQlRequestContext context,
+        String entityType,
+        int entityIdsToFilterSize,
+        List<AttributeAssociation<FilterArgument>> filterArguments,
+        DataFetchingFieldSelectionSet selectionSet) {
+      boolean fetchTotal =
```

Comment:
Even though I am setting fetchTotal to true, the total count is still restricted by the limit. My assumption was limit will be enforced only on the entities being retrieved, and not on the total count.